core/vm/instructions.go

```diff
@@ -529,10 +529,11 @@ func opCreate(instr instruction, pc *uint64, env Environment, contract *Contract
 		dataGas.Mul(dataGas, params.CreateDataGas)
 		if contract.UseGas(dataGas) {
 			env.Db().SetCode(addr, ret)
+			stack.push(addr.Big())
+		} else if params.IsHomestead(env.BlockNumber()) {
+			env.Db().Delete(addr)
+			stack.push(new(big.Int))
 		}
```

Comment:
I believe you're missing a case here; `contract.UseGas(dataGas) == false` and not homestead. Before this commit; that case still pushed address to stack. As I read it right now, you will fail to push the address for pre-homestead contract creation failures. 
I think it should be something like this:

		stack.push(addr.Big())
	}
	else 
	{
		if params.IsHomestead(env.BlockNumber()) {
			env.Db().Delete(addr)
			stack.push(new(big.Int))
		}else{			
			stack.push(addr.Big)
		}
	}